modules/images/webp-uploads/helper.php

```
@@ -65,6 +65,45 @@ function webp_uploads_get_upload_image_mime_transforms() {
  */
 function webp_uploads_generate_additional_image_source( $attachment_id, array $size_data, $mime, $destination_file_name = null ) {
+
+	/**
+	 * Filter to allow the generation of additional image sources, in which a defined mime type
+	 * can be transformed and create additional mime types for the file.
+	 *
+	 * Returning an image data array or WP_Error here effectively short-circuits the default logic to generate the image source.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array|null|WP_Error  $image         Image data {'path'=>string, 'file'=>string, 'width'=>int, 'height'=>int, 'mime-type'=>string} or null or WP_Error.
+	 * @param int                  $attachment_id The ID of the attachment from where this image would be created.
+	 * @param string               $image_size    The size name that would be used to create this image, out of the registered subsizes.
+	 * @param array                $size_data     An array with the dimensions of the image: height, width and crop {'height'=>int, 'width'=>int, 'crop'}.
+	 * @param string               $mime          The target mime in which the image should be created.
+	 *
+	 * @return array|null|WP_Error An array with the file and filesize if the image was created correctly otherwise a WP_Error
+	 */
+	$image = apply_filters( 'webp_uploads_pre_generate_additional_image_source', null, $attachment_id, 'full', $size_data, $mime );
+
+	// Try to check for not empty returned filtered image.
+	if ( ! empty( $image ) ) {
+		if ( is_wp_error( $image ) ) {
+			return $image;
+		}
+
+		if ( empty( $image['file'] ) ) {
+			return new WP_Error( 'image_file_not_present', __( 'The file key is not present on the image data', 'performance-lab' ) );
+		}
+
+		if ( empty( $image['path'] ) || ! file_exists( $image['path'] ) ) {
+			return new WP_Error( 'image_path_not_present', __( 'The file path is not present on the image data', 'performance-lab' ) );
+		}
```

These conditions shouldn't be there, we shouldn't return errors when the filter is being misused by a developer. Your original condition to only return the array below if `is_array( $image ) && ! empty( $image['file'] ) && ! empty( $image['path'] )` was more accurate here.

We should preferably fail silently if someone returns something from the filter that they shouldn't.

---

Fixed now!